### DISCUSSION TITLE
privatebin & nginx proxy manager

### DISCUSSION THREAD

**Discussion by Tealk** [PrivateBin/PrivateBin#933] [Q&A] [answered]
Hello,

I have privatebin on a vserver sitting in a vlan behind pfsense, the nginx proxy manager takes care of the proper routing.
Now when I want to save text I get the message: "Text could not be created: Error on server or no response from server".
Do I have to make any special settings?

**Comment by rugk** [comment, marked as answer]
What's the server log/error?

**Reply by Tealk** [reply]
this is in the nginx error log
```
2022/06/04 11:51:58 [error] 421#421: *2309 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught PDOException: could not find driver in /var/www/privatebin/lib/Data/Database.php:110
Stack trace:
#0 /var/www/privatebin/lib/Data/Database.php(110): PDO->__construct()
#1 [internal function]: PrivateBin\Data\Database::getInstance()
#2 /var/www/privatebin/lib/Model.php(86): forward_static_call()
#3 /var/www/privatebin/lib/Controller.php(199): PrivateBin\Model->getStore()
#4 /var/www/privatebin/lib/Controller.php(125): PrivateBin\Controller->_create()
#5 /var/www/privatebin/index.php(18): PrivateBin\Controller->__construct()
#6 {main}
  thrown in /var/www/privatebin/lib/Data/Database.php on line 110" while reading response header from upstream, client: 192.168.100.11, server: _, request: "POST / HTTP/1.1", upstream: "fastcgi://unix:/run/php/php8.0-fpm.sock:", host: "paste.rollenspiel.monster", referrer: "https://paste.rollenspiel.monster/"
```

**Reply by rugk** [reply]
PDOException -> You try to use the DB for saving data, but PHP has not installed the correct module for your DB.

This is not related to nginx or your routing etc. Just search for the error message or how to setup PHP PDO. See also the installation guide: https://github.com/PrivateBin/PrivateBin/blob/master/INSTALL.md

**Reply by Tealk** [reply]
I was really sure that I had installed all the necessary packages, ok sorry for the topic
